nmap/test/helpers/test_nmap_output_parser.py

```diff
@@ -0,0 +1,112 @@
+from unittest import TestCase
+
+from src.helpers.nmap_output_parser import NmapOutputParser
+
+parse = NmapOutputParser()
+
+
+class NmapOutputParserTest(TestCase):
+    def setUp(self):
+        self.data_assets = {
+            "injection": {"inject_content": {"target_selector": "assets"}}
+        }
+
+        self.data_no_assets = {
+            "injection": {"inject_content": {"target_selector": "other"}}
+        }
+
+        self.result_multiple_hosts = {
+            "nmaprun": {
+                "host": [
+                    {
+                        "address": {"@addr": "10.0.0.1"},
+                        "ports": {
+                            "port": [
+                                {
+                                    "@portid": "22",
+                                    "state": {"@state": "open"},
+                                    "service": {"@name": "ssh"},
+                                },
+                                {
+                                    "@portid": "80",
+                                    "state": {"@state": "closed"},
+                                    "service": {"@name": "http"},
+                                },
+                            ]
+                        },
+                    },
+                    {
+                        "address": {"@addr": "10.0.0.2"},
+                        "ports": {
+                            "port": [
+                                {
+                                    "@portid": "443",
+                                    "state": {"@state": "open"},
+                                    "service": {"@name": "https"},
+                                }
+                            ]
+                        },
+                    },
+                ]
+            }
+        }
+
+        self.result_single_host = {
+            "nmaprun": {
+                "host": {
+                    "address": {"@addr": "172.16.5.10"},
+                    "ports": {
+                        "port": [
+                            {
+                                "@portid": "21",
+                                "state": {"@state": "open"},
+                                "service": {"@name": "ftp"},
+                            }
+                        ]
+                    },
+                }
+            }
+        }
+
+    # -------------------------------
+    # Tests
+    # -------------------------------
+
+    def test_parse_target_assets(self):
+        """Ensure target_selector='assets' uses asset_list and sets asset_id."""
+        data = {"injection": {"inject_content": {"target_selector": "assets"}}}
+
+        result = parse(data, self.result_single_host, ["asset-123"])
```

Comment:
Tests don't run: here `parse` is an instance of `NmapOutputParser`, not a `callable` (i.e. not a function).
```suggestion
        result = parse.parse(data, self.result_single_host, ["asset-123"])
```
You should perhaps review the naming.